### PR TITLE
Simplify and specialise ouroboros-network interface

### PR DESCRIPTION
**Pull request by coot**
* Expose function which are specific to `NdoeToNodeV_1` and `NodeToClientV_1`
* Remove `AnyResponderApp`
* Consolidate exports in `NodeToNode` and `NodeToClient` modules

WIth this patch all the api that consensus needs is contained in `NodeToClient` and `NodeToNode`.  I'd expect the same is true for other applications (wallet, explorer, chairman, etc.)

### CODE REVIEW

**Comment by coot** [comment]
@erikd there are some further changes that we can make.  I was thinking that we could allow for `MuxPeer` to run bracket like operations, this way you could use `MuxPeer` instead `OuroborosApplication` type.   I think this would improve the error you got from GHC.  But this is not included in this PR.

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs [hunk 1]
```diff
@@ -296,51 +288,37 @@ initNetwork registry nodeArgs kernel RunNetworkArgs{..} = do
...
-                                DictVersion
-                                (NetworkApps peer)
-    networkAppNodeToClient =
-      simpleSingletonVersions
-        NodeToClientV_1
-        (NodeToClientVersionData { networkMagic = 0 })
-        (DictVersion nodeToClientCodecCBORTerm)
-        networkApps
+    -- TODO: network magics should be configurable, this gives an early fail if
+    -- one tries to run testnet vs mainnet.
```
Comment (karknu) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: +1 for the comment, we should have a task for it too.

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs [hunk 2]
```diff
@@ -376,10 +351,5 @@ initNetwork registry nodeArgs kernel RunNetworkArgs{..} = do
       (Just (Socket.SockAddrInet6 0 0 (0, 0, 0, 1) 0))
```
Comment (karknu) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Not related to this PR, but the IPv6 address looks wrong. ::1 is the loop back address, IPv6's version of 127.0.0.1.

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs (hunk 1 above)
Comment (coot) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Exactly, we need to bug `consensus` team for that :).

File: ouroboros-consensus/src/Ouroboros/Consensus/Node.hs (hunk 2 above)
Comment (coot) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: @karknu aren't they the same:
```
Prelude Network.Socket> SockAddrInet6 0 0  (0,0,0,1) 0
[::1]:0
```

Comment (coot) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node.hs]: Ach right, I see what you mean, it should be `(0,0,0,0)` here.

**Pull request merged**